crates/ty_project/src/metadata/options.rs

```diff
@@ -277,14 +278,36 @@ impl Options {
             roots
         };
 
-        let settings = SearchPathSettings {
-            extra_paths: environment
+        // collect the existing site packages
+        let mut extra_paths: Vec<SystemPathBuf> = Vec::new();
+
+        // read all the paths off the PYTHONPATH environment variable, check
+        // they exists, and add them to the vec of extra_paths as they should
+        // be checked before site-packages just like python interpreter does
+        if let Ok(python_path) = env::var("PYTHONPATH") {
+            let splits: Vec<SystemPathBuf> = python_path
+                .split(':')
+                .filter_map(|str_path| {
+                    let possible_path = std::path::PathBuf::from(str_path);
+                    match possible_path.try_exists() {
+                        Ok(true) => SystemPathBuf::from_path_buf(possible_path).ok(),
+                        _ => None,
+                    }
+                })
+                .collect();
+            extra_paths.extend(splits);
```

Comment:
It's not really necessary to collect before calling `extend`. You can pass the iterator directly. However, as mentioned by @carljm on the issue, we should inform users that ty's adding paths from `PYTHONPATH`.

That's why I would rewrite this as:

```rust
for path in python_path.split(':') {
	let possible_path = SystemPathBuf::from(str_path);

	if system.path_exists(&possible_path) {
		tracing::debug!("Adding `{possible_path}` from the `PYTHONPATH` environment variable to `extra_paths`);
		extra_paths.push(possible_path);
	} else {
		tracing::debug!("Skipping `{possible_path}` listed in `PYTHONPATH` because the path doesn't exist)"
	}
}
```

We may even want